srcs/app/utils/SyscallWrap.cpp

```diff
@@ -63,7 +68,14 @@ pid_t SyscallWrap::forkWr(const std::string &file,
 						  std::size_t line) {
 	int ret;
 	if ((ret = fork()) == -1) {
+#ifdef DBG
 		AddDebuggingInfo_(file, "fork", func, line);
+#else
+		(void) file;
+		(void) func;
+		(void) line;
+		throw std::runtime_error(std::strerror(errno));
+#endif
```

Comment:
What do you think about adding this `#ifdef`/`#else`/`#endif` only in `AddDebuggingInfo_()` method? Also, maybe renaming it to something like `ThrowException_()` or something like that.

We could also use the `dbgargs()` macro here.

Reply:
well, the dbg macro needs to expand, to something or to nothing. Otherwise it will be an unknown symbol. But it needs to exist in either debugging or regular build. How would you make that work without the #ifdef #else #endif that controls the (yet to define name, but for now) dbg() macro definition?

Reply:
I realize that we can't use the `DEBUG_ARGS` macro here because it starts with a comma, sorry.

Something like this, we would need a new macro. `DEBUG_ARGS_NAMES`?
```
# define dbgargs_names() , file, func, line
```
```
ThrowException_("name" DEBUG_ARGS_NAMES);

```

```
void SyscallWrap::ThrowException_(const std::string &sys_call_name DEBUG_ARGS) {
	std::ostringstream ss;
	ss << sys_call_name << ": " << std::strerror(errno)
#ifdef DBG
		<< ", in " << file << ":"<< line
		<< ":" << func << "()"
#endif
		;
	throw std::runtime_error(ss.str());
}

```

Reply:
Done